core/src/main/java/org/apache/gravitino/credential/CredentialUtils.java

```diff
@@ -19,14 +19,74 @@
 
 package org.apache.gravitino.credential;
 
+import com.google.common.base.Splitter;
 import com.google.common.collect.ImmutableSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
 import org.apache.gravitino.utils.PrincipalUtils;
 
 public class CredentialUtils {
+
+  private static final Splitter splitter = Splitter.on(",");
+
   public static Credential vendCredential(CredentialProvider credentialProvider, String[] path) {
     PathBasedCredentialContext pathBasedCredentialContext =
         new PathBasedCredentialContext(
             PrincipalUtils.getCurrentUserName(), ImmutableSet.copyOf(path), ImmutableSet.of());
     return credentialProvider.getCredential(pathBasedCredentialContext);
   }
+
+  public static Map<String, CredentialProvider> loadCredentialProviders(
+      Map<String, String> catalogProperties) {
+    Set<String> credentialProviders =
+        CredentialUtils.getCredentialProvidersByOrder(() -> catalogProperties);
+
+    return credentialProviders.stream()
+        .collect(
+            Collectors.toMap(
+                String::toString,
+                credentialType ->
+                    CredentialProviderFactory.create(credentialType, catalogProperties)));
+  }
+
+  /**
+   * Get Credential providers from properties supplier.
+   *
+   * <p>If there are multiple properties suppliers, will try to get the credential providers in the
+   * input order.
+   *
+   * @param propertiesSuppliers The properties suppliers.
+   * @return A set of credential providers.
+   */
+  public static Set<String> getCredentialProvidersByOrder(
+      Supplier<Map<String, String>>... propertiesSuppliers) {
+
+    for (Supplier<Map<String, String>> supplier : propertiesSuppliers) {
+      Map<String, String> properties = supplier.get();
+      Set<String> providers = getCredentialProvidersFromProperties(properties);
+      if (!providers.isEmpty()) {
+        return providers;
+      }
+    }
+
+    return ImmutableSet.of();
+  }
+
+  private static Set<String> getCredentialProvidersFromProperties(Map<String, String> properties) {
+    if (properties == null) {
+      return ImmutableSet.of();
```

Comment:
We can use `Collections.emptySet()` for here and other places.

Reply:
updated